internal/cli/cli.go

```diff
@@ -11,9 +11,26 @@ func PromptInput() (string, error) {
 	// output := flag.String("o", "output_file", "Output file path (default: output_file)")
 	flag.Parse()
 
+<<<<<<< HEAD
+	// Error Handling
+
+	if url == "" {
+		return "", errors.New("url can't be empty")
+	}
+
+	if len(url) < 8 || url[:8] != "https://" {
```

Comment:
why length of `url` should be less then 8? There should be no limit on length. For example: http://www.google.com ... or... http://www.xyz.com/download/example/file

There should be no limit on entire length of `url`

Reply:
Rest of `http:` logic is fine.

Reply:
The current checking the first 8 characters for "https://" will cause a runtime panic if the URL is shorter than 8 characters. This is because slicing a string beyond its length is not safe. Without a length check, this validation won't be triggered properly for shorter URLs, leading to potential errors.

Reply:
Fix URL validation by adding strings package. Fixed